lib/features/mailbox_dashboard/presentation/widgets/search_filters/filter_message_button.dart

```diff
@@ -0,0 +1,169 @@
+              onTapActionCallback: () => onDeleteFilterMessageOptionAction?.call(filterMessageOption),
+            ),
+        ]
+      )
+    );
+
+    if (onSelectFilterMessageOptionAction == null) {
+      return childItem;
+    }
+
+    if (!isSelected) {
+      return Material(
+        type: MaterialType.transparency,
+        child: InkWell(
+          onTapDown: (details) => _onTapDownAction(context, details),
+          borderRadius: FilterMessageButtonStyle.borderRadius,
+          child: childItem
+        ),
+      );
+    }
+
+    return Container(
+      decoration: BoxDecoration(
+        borderRadius: FilterMessageButtonStyle.borderRadius,
+        color: filterMessageOption.getBackgroundColor(isSelected: isSelected)),
+      padding: FilterMessageButtonStyle.getButtonPadding(isSelected),
+      child: Row(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          InkWell(
+            onTapDown: (details) => _onTapDownAction(context, details),
+            borderRadius: FilterMessageButtonStyle.borderRadius,
+            child: Row(
+              mainAxisSize: MainAxisSize.min,
+              children: [
+                SvgPicture.asset(
+                  filterMessageOption.getIcon(imagePaths),
+                  width: FilterMessageButtonStyle.iconSize,
+                  height: FilterMessageButtonStyle.iconSize,
+                  colorFilter: filterMessageOption.getIconColor().asFilter(),
+                  fit: BoxFit.fill
+                ),
+                const SizedBox(width: FilterMessageButtonStyle.spaceSize),
+                Text(
+                  filterMessageTitle.length > _titleCharactersMaximum
+                    ? '${filterMessageTitle.substring(0, _titleCharactersMaximum)}...'
+                    : filterMessageTitle,
+                  maxLines: 1,
+                  overflow: TextOverflow.ellipsis,
+                  style: FilterMessageButtonStyle.titleStyle,
+                ),
+                Padding(
+                  padding: FilterMessageButtonStyle.elementPadding,
+                  child: SvgPicture.asset(
+                    imagePaths.icDropDown,
+                    width: FilterMessageButtonStyle.arrowDownIconSize,
+                    height: FilterMessageButtonStyle.arrowDownIconSize,
+                    colorFilter: AppColor.colorFilterMessageIcon.asFilter(),
+                    fit: BoxFit.fill
+                  ),
+                ),
+              ],
+            ),
+          ),
+          TMailButtonWidget.fromIcon(
+            icon: imagePaths.icDeleteSelection,
+            iconSize: FilterMessageButtonStyle.deleteIconSize,
+            iconColor: AppColor.colorFilterMessageIcon,
+            padding: const EdgeInsets.all(8),
+            backgroundColor: Colors.transparent,
+            onTapActionCallback: () => onDeleteFilterMessageOptionAction?.call(filterMessageOption),
+          ),
+        ]
+      )
+    );
```

Comment:
This `return` widget is almost the same as `childItem`. Can we update it somehow to reduce code duplication?

Reply:
It has a slightly different widget layout for each case. We can only optimize it by separating the widget components into stateless ones.

Reply:
Optimized